src/sqlparser/src/ast/value.rs

```diff
@@ -64,6 +65,7 @@ impl fmt::Display for Value {
             Value::SingleQuotedString(v) => write!(f, "'{}'", escape_single_quote_string(v)),
             Value::NationalStringLiteral(v) => write!(f, "N'{}'", v),
             Value::HexStringLiteral(v) => write!(f, "X'{}'", v),
+            Value::CstyleEscapesString(v) => write!(f, "E'{}'", v),
```

Comment:
Just though of another issue 🤦, about unparsing.

Try to create a new yaml file under `src/sqlparser/tests/testdata`:
```
- input: SELECT 'a''b'
  formatted_sql: SELECT 'a''b'

- input: SELECT e'a''b'
  formatted_sql: SELECT E'a''b'
```

The actual unparse result would produce an invalid query: `SELECT E'a'b'`.

Note that unparsing does not need to reproduce the original query, but it should produce an equivalent one. To be more specific, it is okay to unparse `e'a\'b''c'` into `e'a''b''c'` or `e'a\'b\'c'`.

No matter which of the 3 styles above you choose, make sure to test the unparsing of `\\` as well. For example:
```
select e'\\', e'\'', e'\\\'', e'\\\'''', e'\z';
```

Reply:
Maybe the easiest fix is to push `\'` when `tokenize_single_quoted_string_with_escape` sees `''`.

Rationale:
* The `SingleQuotedString` converts `''` to `'` during parsing and reverts it during unparsing.
* The `CstyleEscapesString` attempts to preserve escape sequences until binding, to make unparsing easy. But `''` is being treated the same way as in `SingleQuotedString` during parsing.